app/controllers/api/v1/questionnaires_controller.rb

```diff
@@ -0,0 +1,107 @@
+  def index
+    @questionnaires = Questionnaire.order(:id)
+    render json: @questionnaires, status: :ok and return
+  end
+  
+  # Show method returns the JSON object of questionnaire with id = {:id}
+  # GET on /questionnaires/:id
+  def show
+    begin
+      @questionnaire = Questionnaire.find(params[:id])
+      render json: @questionnaire, status: :ok and return
+    rescue ActiveRecord::RecordNotFound
+      render json: $ERROR_INFO.to_s, status: :not_found and return
+    end
+  end
+  
+  # Create method creates a questionnaire and returns the JSON object of the created questionnaire
+  # POST on /questionnaires
+  # Instructor Id statically defined since implementation of Instructor model is out of scope of E2345.
+  def create
+    begin
+      @questionnaire = Questionnaire.new(questionnaire_params)
+      @questionnaire.display_type = sanitize_display_type(@questionnaire.questionnaire_type)
+      @questionnaire.save!
+      render json: @questionnaire, status: :created and return
+    rescue ActiveRecord::RecordInvalid
+      render json: $ERROR_INFO.to_s, status: :unprocessable_entity
+    end
+  end
+
+  # Destroy method deletes the questionnaire object with id- {:id}
+  # DELETE on /questionnaires/:id
+  def destroy
+    begin
+      @questionnaire = Questionnaire.find(params[:id])
+      @questionnaire.delete
+    rescue ActiveRecord::RecordNotFound
+      render json: $ERROR_INFO.to_s, status: :not_found and return
+    rescue ActiveRecord::RecordInvalid
+      render json: $ERROR_INFO.to_s, status: :unprocessable_entity
+    end
+  end
+
+  # Update method updates the questionnaire object with id - {:id} and returns the updated questionnaire JSON object
+  # PUT on /questionnaires/:id
+  def update
+    # Save questionnaire information
+    begin
+      @questionnaire = Questionnaire.find(params[:id])
+      @questionnaire.update(questionnaire_params)
+      @questionnaire.save!
+      render json: @questionnaire, status: :ok and return
+    rescue ActiveRecord::RecordNotFound
+      render json: $ERROR_INFO.to_s, status: :not_found and return
+    rescue ActiveRecord::RecordInvalid
+      render json: $ERROR_INFO.to_s, status: :unprocessable_entity
+    end
+  end
+
+  # Copy method creates a copy of questionnaire with id - {:id} and return its JSON object
+  # POST on /questionnaires/copy/:id
+  def copy
+    begin
+      @questionnaire = Questionnaire.copy_questionnaire_details(params)
+      render json: "Copy of the questionnaire has been created successfully.", status: :ok and return
+    rescue ActiveRecord::RecordNotFound
+      render json: $ERROR_INFO.to_s, status: :not_found and return
+    rescue ActiveRecord::RecordInvalid
+      render json: $ERROR_INFO.to_s, status: :unprocessable_entity
+    end
+  end
+
+  # Toggle access method toggles the private variable of the questionnaire with id - {:id} and return its JSON object
+  # GET on /questionnaires/toggle_access/:id
+  def toggle_access
```

Comment:
The code looks good, but you can simplify it a bit using a ternary operator for setting the @access variable:

```ruby
def toggle_access
  begin
    @questionnaire = Questionnaire.find(params[:id])
    @questionnaire.toggle!(:private)
    @access = @questionnaire.private ? 'private' : 'public'
    render json: "The questionnaire \"#{@questionnaire.name}\" has been successfully made #{@access}. ",
      status: :ok
  rescue ActiveRecord::RecordNotFound
    render json: $ERROR_INFO.to_s, status: :not_found
  rescue ActiveRecord::RecordInvalid
    render json: $ERROR_INFO.to_s, status: :unprocessable_entity
  end
end
```
Here, the `toggle!` method is used to toggle the boolean value of the private attribute of the `@questionnaire` object. Also, the ternary operator is used to set the `@access` variable based on the current value of the private attribute.

Reply:
Commit e86e810 addresses this review.